Foreman/DataCache.cs

```diff
@@ -1084,58 +1093,108 @@ private static Item FindOrCreateUnknownItem(String itemName)
         private static Bitmap GetIcon(LuaTable values, bool fallbackToUnknown = false)
         {
             string fileName = ReadLuaString(values, "icon", true);
-            if (fileName == null)
+            string mipmaps = ReadLuaString(values, "icon_mipmaps", true);
+            int iconsize = 0;
+            if (mipmaps != null)
             {
-                LuaTable icons = ReadLuaLuaTable(values, "icons", true);
+                iconsize = 4 * Convert.ToInt32(Math.Pow(2, Convert.ToDouble(mipmaps)));
+            }
 
-                try
+
+
+            return (fileName == null ? GetIcons(values) : LoadImage(fileName, iconsize));
+        }
+        private static Bitmap GetIcons(LuaTable values)
+        {
+            Bitmap fullImage = new Bitmap(32,32);
+            for (int i = 0; i < 32; i++)
+            {
+                for (int j = 0; j < 32; j++)
                 {
-                    if (!Convert.ToString(values["name"]).Contains("barrel"))
+                    fullImage.SetPixel(i, j, Color.FromArgb(0, 255, 255, 255));
+                }
+            }
+            LuaTable icons = ReadLuaLuaTable(values, "icons", true);
+            try
+            {
+                if (!Convert.ToString(values["name"]).Contains("barrel"))

```

Comment:
I agree barrels are problematic and cause long load times but this is not the place to exclude them. I already have some commits pushed that attempt to "disable" barreling recipes maybe we can also exclude the items there as well.

Reply:
well, if you have something in the pipeline already which would make this obsolete, go for it, otherwise i would keep it for now, because it greatly reduces load time atm^^

Reply:
Fair enough I'll create an issue to track disabling or excluding barrels in general and reference this code to be removed once we have a general solution.